documentation/Using-the-SDK/Creating-a-pipeline.md

```diff
@@ -2,58 +2,62 @@
-1) Data Source Parser
-2) One or more Data Cookers (Cookers)
+Data Processing Pipelines (DPPs) allow you to compose components together that 
+map data between inputs and outputs. A DPP is comprised of the following:
+1) one __source parser__
+2) one or more `DataCooker`s (cookers)
 
-Cookers can be further broken down into:
-1) Source Cookers
-2) Composite Cookers
+For a real world example of a pipeline, see the LTTng SDK plugin, found on GitHub [here](https://github.com/microsoft/Microsoft-Performance-Tools-Linux)
 
-All Cookers expose one or more __Data Outputs__. Data Outputs expose data that
-can be consumed by users of the DPP or other Cookers in the DPP.
+Cookers can be further broken down into: 
+1) `SourceCooker`s
+2) `CompositeCooker`s
 
-__Source Cookers__ take data directly from a Data Source Parser and produce
-__Data Outputs__. __Composite Cookers__ take data from other cookers to produce
-data outputs. A Composite Cooker may depend on both __Source__ and
- __Composite__ cookers.
+`SourceCooker`s take data directly from a source parser and produce 
+`DataOutput`s. `CompositeCooker`s take data from other cookers to produce 
+`DataOutput`s. A `CompositeCooker` may depend on both __source__ *and* 
+ __composite__ cookers.
 
 The following illustrates these concepts:
 
-![Pipeline](.attachments/dpp.png)
+<img src=".attachments/dpp.svg" width="500">
 
-For a real example, see the LTTng Plugin for WPA, found [Here](https://github.com/microsoft/Microsoft-Performance-Tools-Linux)
+All cookers expose zero or more __data outputs__. `DataOutput`s expose data that 
+can be consumed by users of the DPP or other cookers in the DPP. 
 
-## Data Source Parsers
+## Source Parsers
 
-__Data Source Parsers__ parse data from a Data Source into data that can be
-manipulated by your application. For example, parsing an ETW ETL file into a
-stream of Event objects.  The following are required in order to implement a
-Data Source Parser:
-1) A Custom Data Source
-2) A class implementing SourceParserBase
-3) A Custom Data Processor implementing CustomDataProcessorBaseWithSourceParser
+__Source parsers__ parse data from a data source into data that can be 
+manipulated by your application. For example, a source parser may parse an ETW 
+ETL file into a stream of `Event` objects.  The following are required in order to implement a 
+source parser:
+1) A `CustomDataSource`
+2) A class implementing `SourceParserBase`
+3) A `CustomDataProcessor` implementing `CustomDataProcessorBaseWithSourceParser`
 
-A __Custom Data Source__ (1) is required in order to expose your data, whether
-you are using DPPs or not. The Custom Data Source is used as the entry point for
-creating Custom Data Processors for processing your data. Please see
-[here](/Creating-a-simple-custom-data-source) for more on Custom Data Sources.
+A `CustomDataSource` is required in order to expose your data, whether 
+you are using DPPs or not. The `CustomDataSource` is used as the entry point for 
```

Comment:
If you make the previous change (from DPPs -> DPP) also change it here and make it singular.